synapse/config/workers.py

```diff
@@ -164,7 +213,37 @@ def read_config(self, config, **kwargs):
                 "Must only specify one instance to handle `receipts` messages."
             )
 
-        self.events_shard_config = ShardedWorkerHandlingConfig(self.writers.events)
+        if len(self.writers.events) == 0:
+            raise ConfigError("Must specify at least one instance to handle `events`.")
```

Comment:
I wonder if it would help us remember to add these if we used `validators` on the `WriterLocations` attributes?

Reply:
Yeah, though I'm not sure how easy it is to make `ConfigError` out of validation failures?

Reply:
There's a few places that we raise a `ConfigError` from another exception, but not sure if it is worth refactoring that here. 🤷